MyFridge/MyFridge/Shelf.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyFridge
+{
+    internal class Shelf
+    {
+        public static int UniqueId { get; set; } = 1;
+        public int Id { get; }
+        public int FloorNumber { get; }
+        private int _spaceAvailable = 400;
+
+        public int SpaceAvailable
+        {
+            get => _spaceAvailable; set
+            {
+                if (value <= 0)
+                {
+                    throw new Exception("There is no space");
+                }
+
+                _spaceAvailable = value;
+            }
+        }
+        public List<Item> Items { get; set;}
+
+
+        public Shelf(int floorNumber, List<Item> items)
+        {
+            
+            Id += UniqueId++;
+
+            FloorNumber = floorNumber;
+                //if (items != null)
+                //{
+                //    foreach (var item in items)
+                //    {
+                //        if (item == null) items.Remove(item);
+                //    }
+                foreach (var item in items)
+                {
+                    if (_spaceAvailable - item.Space >= 0)
+                        _spaceAvailable -= item.Space;
+                    else
+                        items.Remove(item);
```

Comment:
modifying a list while iterating it is not recommended as it can lead to unexpected behavior.
you can accumulate the space of all products and then decide if you accept the whole input or not.
I suggest not accepting only part of the input (and removing some items) as this behavior is not clear to the calling method (which items you choose to remove? Does order matter? etc).